src/flow/src/adapter.rs

```diff
@@ -109,7 +112,28 @@ impl Default for FlownodeOptions {
     }
 }
 
-impl Configurable for FlownodeOptions {}
+impl Configurable for FlownodeOptions {
+    fn validate(&self) -> common_config::error::Result<()> {
+        use common_config::error::LoadLayeredConfigSnafu;
+        if self.num_workers == 0 {
+            Err(ConfigError::Message(
+                "num_workers must be at least 1".to_string(),
+            ))
+            .context(LoadLayeredConfigSnafu)?
+        }
+        if self.num_workers > MAX_WORKERS {
+            Err(ConfigError::Message(format!(
+                "num_workers cannot exceed {}",
+                MAX_WORKERS
+            )))
+            .context(LoadLayeredConfigSnafu)?
+        }
+        Ok(())
+    }
+}
+
+/// Define a reasonable maximum number of workers
+const MAX_WORKERS: usize = 16;
```

Comment:
Looks like it's not necessary. For example, tokio runtime doesn't limit the worker threads number.